pkg/nodeconfig/payload/payload.go

```diff
@@ -338,3 +391,27 @@ func generateNetworkConfigScript(clusterCIDR, hnsNetworkName, hnsPSModulePath,
 	}
 	return networkConfScript, nil
 }
+
+// Creates a .tar.gz archive from file data
+func createTarGzFile(data []byte, fileName string, outWriter io.Writer) error {
+	// Chain writers: File -> Gzip -> Tar
+	gzipWriter := gzip.NewWriter(outWriter)
```

Comment:
consider using func `NewWriterLevel(w io.Writer, level int) (*Writer, error)`   with `BestCompression`

https://pkg.go.dev/compress/gzip#BestCompression

Reply:
actually we can do  "Level 1 (Fastest)" for testing and CI,
and "Level 9 (Best Compression)" for final release.